packages/git/src/browser/git-base-widget.ts

```diff
@@ -53,4 +103,56 @@ export class GitBaseWidget extends VirtualWidget {
         }
         return result;
     }
+
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        this.addKeyListener(this.node, Key.ARROW_LEFT, () => this.handleLeft());
+        this.addKeyListener(this.node, Key.ARROW_RIGHT, () => this.handleRight());
+        this.addKeyListener(this.node, Key.ARROW_UP, () => this.handleUp());
+        this.addKeyListener(this.node, Key.ARROW_DOWN, () => this.handleDown());
+        this.addKeyListener(this.node, Key.ENTER, () => this.handleEnter());
+    }
+
+    protected handleLeft(): void {
+        this.selectNodeByDirection(SelectDirection.PREVIOUS);
+    }
+
+    protected handleRight(): void {
+        this.selectNodeByDirection(SelectDirection.NEXT);
+    }
+
+    protected handleUp(): void {
+        this.selectNodeByDirection(SelectDirection.PREVIOUS);
+    }
+
+    protected handleDown(): void {
+        this.selectNodeByDirection(SelectDirection.NEXT);
+    }
+
+    protected handleEnter(): void {
+
+    }
+
+    protected getSelected(): T | undefined {
+        return this.gitNodes ? this.gitNodes.find(c => c.selected || false) : undefined;
+    }
+
+    protected selectNode(node: T) {
+        const n = this.getSelected();
+        if (n) {
+            n.selected = false;
+        }
+        this.selectionService.selection = node;
+    }
+
+    protected selectNodeByDirection(dir: SelectDirection) {
```

Comment:
Why don't have 2 methods instead `selectNextNode` and `selectPrevNode`? less to type on the callee side, no need for `SelectDirection`